src/App.js

```diff
@@ -17,22 +17,28 @@ export class App extends Component {
     route: PropTypes.oneOfType([
       PropTypes.func,
       PropTypes.element
-    ]).isRequired
+    ]).isRequired,
+    ipfsReady: PropTypes.bool.isRequired,
+    ipfsInitFailed: PropTypes.bool.isRequired
   }
 
   componentWillMount () {
     this.props.doInitIpfs()
   }
 
   render () {
-    const Page = this.props.route
+    const { route: Page, ipfsReady, ipfsInitFailed } = this.props
+    // Only shows the page if IPFS is ready of if the initialization has failed.
+    // This way we can make sure we always use user's own node if available and
+    // the public gatway otherwise.
+    const ready = ipfsReady || ipfsInitFailed
 
     return (
       <div className='App sans-serif' onClick={navHelper(this.props.doUpdateUrl)}>
         <div className='flex flex-column min-vh-100'>
           <Header />
           <main className='flex-auto pa4'>
-            <Page />
+            { ready && <Page /> }
```

Comment:
Not sure about this, if it takes a bit to init ipfs the page will be blank with nothing to show. Do you think this is the best solution?

Reply:
@fsdiogo we're just using window.ipfs or js-ipfs-api and then requesting the ID. It should not take much time, but we can add a loading animation. What do you think?

Reply:
Alright, we can do that later in another PR!